src/Microsoft.Azure.WebJobs.Host/Config/ServiceProviderWrapper.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.WebJobs.Host.Config
+{    
+    // Used to wrap a JobHostConfiguration. Let's JobHost startup modify services  
+    // without modifying the underlying configuration. 
+    internal class ServiceProviderWrapper : IServiceProvider
```

Comment:
my one reservation was that this class is totally generic and has no connection to JobHost.